### DISCUSSION TITLE
Allow default headers for GET methods

### DISCUSSION THREAD

**Discussion by ritikexecfn** [O365/python-o365#1021] [Q&A] [answered]
Why don't we allow default headers for prefer type immutable when getting messages?

<img width="480" alt="image" src="https://github.com/O365/python-o365/assets/141752679/499df44a-a195-4555-a312-9c8bfc8f4b66">

**Comment by alejcas** [comment]
can elaborate a little bit more?

**Reply by ritikexecfn** [reply]
I want to get list of messages from mailbox which is a GET request. How can I pass prefer: IdType:Immutable header? This is only allowed in post, put, patch requests as of now

**Reply by alejcas** [reply, marked as answer]
You are correct. I have just committed 464c3e1b856a5e274e8c2325d83b5012692645c5 to allow this.

Can you check that with this code all works as expected?

**Reply by ritikexecfn** [reply]
Working fine now. Thank you.

**Comment by Invincibear** [comment]
@ritikexecfn can you show me an example of using this? I need to use `IdType:Immutable` as well and am not sure how to set this globally.

**Reply by Invincibear** [reply]
I was able to do so by attaching to my instance of Schedule() when using the Calendar API.
```
schedule = account.schedule()
schedule.con.default_headers['Prefer'] = 'IdType="ImmutableId"'
```
